app/models/user.rb

```diff
@@ -1,6 +1,63 @@
 class User < ActiveRecord::Base
 
-  def fullname
-    first_name + " " + last_name
+  #----------------------Server Side Validations--------------------------------
+  validates :first_name, :presence => true, :length => { :maximum => 250 }
+  validates :last_name, :presence => true, :length => { :maximum => 250 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, :presence => true, :format => { :with => VALID_EMAIL_REGEX }, :uniqueness => {:case_sensitive => false}
+  validates :gender, :presence => true, :inclusion => { in: %w(male female)}
+  validates :status, :presence => true, :inclusion => { in: %w(shaved massaged_and_trimmed trimmed unshaved not_applicable)}
+  validate :can_be_marked_unshaved
+  validate :validate_shaving_need_based_on_gender
+  validate :validate_needs_shave_or_trimming
+
+  def can_be_marked_unshaved
+    if !self.needs_shave_or_trimming && has_beard?(status)
+      errors.add(:base, "User can't be marked unshaved as he has shaved / trimmed #{ActionController::Base.helpers.time_ago_in_words(self.shaved_at)} ago.") if self.status_changed?
+    end
+  end
+
+  def full_name
+    (first_name + " " + last_name).titleize
+  end
+
+  def validate_needs_shave_or_trimming
+    if !has_beard?(status_was) && !has_beard?(status) && self.status_changed?
+      errors.add(:base, "User can't be shaved / trimmed again as he has shaved / trimmed #{ActionController::Base.helpers.time_ago_in_words(self.shaved_at)} ago.") if !self.needs_shave_or_trimming
```

Comment:
I am really sorry for this, I'll surely reduce the line size to increase its readability.